os_builders/roles/vm_baseline/tasks/wazuh.yml

```diff
@@ -0,0 +1,135 @@
+- name: Install wazuh on Ubuntu
+  block:
+    - name: Add GPG Key for Wazuh
+      ansible.builtin.apt_key:
+        url: "https://packages.wazuh.com/key/GPG-KEY-WAZUH"
+        state: present
+
+    - name: Add Wazuh repo
+      ansible.builtin.apt_repository:
+        repo: "deb https://packages.wazuh.com/4.x/apt/ stable main"
+        state: present
+
+    - name: Install Wazuh agent
+      ansible.builtin.apt:
+        name: wazuh-agent=4.8.*
+        state: present
+        allow_downgrade: true
+        update_cache: yes
+  when: ansible_distribution == "Ubuntu"
+
+- name: Quattor package name
+  set_fact:
+    quattor_package_name: "ccm"
+
+- name: Install wazuh on Rocky
+  block:
+    - name: add wazuh repo
+      yum_repository:
+        name: wazuh
+        file: wazuh
+        description: "Repository to install wazuh"
+        baseurl: https://packages.wazuh.com/4.x/yum/
+        enabled: true
+        gpgcheck: false
+
+    - name: Install wazuh
+      yum:
+        name:
+          - wazuh-agent-4.8.*
+        state: present
+        disable_gpg_check: true
+  when: ansible_distribution == "Rocky" and "ccm" not in ansible_facts.packages
+
+- name: configure wazuh-agent
+  block:
+  - name: build wazuh groups list
+    set_fact:
+      wazuh_groups:
+        - "cloud"
+        - "{{ ansible_distribution | lower }}"
+        - "{{ ansible_facts['os_family'] | lower}}"
+  - name: build wazuh extra json
+    set_fact:
+      wazuh_extra_json:
+        groups: wazuh_groups
+
+  - name: Ensure wazuh-agent extra directory exists
+    file:
+      path: "/var/ossec/etc/extra"
+      state: directory
+      owner: root
+      group: root
+      mode: 0644
+
+  - name: write out wazuh extra config
+    copy:
+      dest: /var/ossec/etc/extra/01-ansible-baseline.json
+      content: "{{ wazuh_extra_json }}"
+
+  - name: Copy wazuh client enrichment script
+    copy:
+      src: "wazuh/enrich-wazuh-agent.py"
+      dest: "/var/ossec/bin/enrich-wazuh-agent.py"
+      owner: root
+      group: wazuh
+      mode: 0750
```

Comment:
Nit: we lock users out from reading this file even though the parent dir is read-only

Could we do something like: 0754 - so users can see exactly what is running without having the exec bit?

Reply:
Done